api/validation/schemas.py

```diff
@@ -53,72 +51,72 @@ class GetInstanceTypesSchema(Schema):
     region = fields.String(required=True, validate=aws_region_validator)
 
 QueueStatus = QueueStatusSchema(unknown=INCLUDE)
 
 
 class ScontrolJobSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    job_id = fields.String(required=True)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
+    job_id = fields.String(required=True, validate=validate.Length(max=256))
     region = fields.String(required=True, validate=aws_region_validator)
 
 ScontrolJob = ScontrolJobSchema(unknown=INCLUDE)
 
 
 class CancelJobSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    job_id = fields.String(required=True)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
+    job_id = fields.String(required=True, validate=validate.Length(max=256))
     region = fields.String(required=True, validate=aws_region_validator)
 
 CancelJob = CancelJobSchema(unknown=INCLUDE)
 
 
 class SacctSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    cluster_name = fields.String(required=True, validate=is_alphanumeric_with_hyphen)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=32))
+    cluster_name = fields.String(required=True, validate=validate.And(is_alphanumeric_with_hyphen, validate.Length(max=60)))
     region = fields.String(required=True, validate=aws_region_validator)
 
 Sacct = SacctSchema(unknown=INCLUDE)
 
 
 class LoginSchema(Schema):
-    code = fields.String(required=True)
+    code = fields.String(required=True, validate=validate.Length(max=128))
 
 Login = LoginSchema(unknown=INCLUDE)
 
 
 class PushLogSchema(Schema):
     class PushLogEntrySchema(Schema):
         level = fields.String(required=True, validate=valid_api_log_levels_predicate)
-        message = fields.String(required=True)
+        message = fields.String(required=True, validate=validate.Length(max=246000)) # CW limit is 256k, leaving 1k to extra and level
         extra = fields.Dict()
 
     logs = fields.List(fields.Nested(PushLogEntrySchema), required=True)
 
 PushLog = PushLogSchema(unknown=INCLUDE)
 
 class PriceEstimateSchema(Schema):
-    cluster_name = fields.String(required=True)
-    queue_name = fields.String(required=True)
+    cluster_name = fields.String(required=True, validate=validate.Length(max=60))
+    queue_name = fields.String(required=True, validate=validate.Length(max=60))
     region = fields.String(validate=aws_region_validator, required=True)
 
 PriceEstimate = PriceEstimateSchema(unknown=INCLUDE)
 
 class PCProxySchema(Schema):
-    path = fields.String(required=True)
+    path = fields.String(required=True, validate=validate.Length(max=512))
```

Comment:
question: is this for the YAML configuration? If so I think 512 is too low

Reply:
This is for the path to use to invoke the PC API
512 is large in this case but I put 512 since I have no way of anticipating how long an URL for PC APIs will be
I believe it could be 256 and we'd still be safe